src/hazelcore/ZExp.re

```diff
@@ -876,31 +1027,43 @@ and move_cursor_right_zoperator =
 and move_cursor_right_zoperand =
   fun
   | z when is_after_zoperand(z) => None
+
   | CursorE(OnOp(_), _) => None
+
+  | CursorE(OnText(j), StringLit(_, s) as operand)
+      when j == String.length(s) =>
+    Some(CursorE(OnDelim(1, Before), operand))
   | CursorE(OnText(j), e) => Some(CursorE(OnText(j + 1), e))
+
   | CursorE(OnDelim(k, Before), e) => Some(CursorE(OnDelim(k, After), e))
   | CursorE(OnDelim(_, After), EmptyHole(_) | ListNil(_)) => None
-  | CursorE(OnDelim(_k, After), Parenthesized(body)) =>
-    // _k == 0
+  | CursorE(OnDelim(_zero, After), StringLit(_) as operand) =>
+    Some(CursorE(OnText(0), operand))
+  | CursorE(OnDelim(_zero, After), Parenthesized(body)) =>
     Some(ParenthesizedZ(place_before(body)))
-  | CursorE(OnDelim(_k, After), Inj(err, side, body)) =>
-    // _k == 0
+  | CursorE(OnDelim(_zero, After), Inj(err, side, body)) =>
     Some(InjZ(err, side, place_before(body)))
-  | CursorE(OnDelim(k, After), Lam(err, arg, ann, body)) =>
-    // k == 0 || k == 1 || k == 2
-    switch (k == 0, k == 1, ann) {
-    | (true, _, _) => Some(LamZP(err, ZPat.place_before(arg), ann, body))
-    | (_, true, None) =>
-      // invalid cursor position
-      None
-    | (_, true, Some(ann)) =>
-      Some(LamZA(err, arg, ZTyp.place_before(ann), body))
-    | (false, false, _) => Some(LamZE(err, arg, ann, place_before(body)))
+  | CursorE(OnDelim(k, After), Lam(err, arg, ann, body)) => {
```

Comment:
don't need this diff -- better to avoid making formatting changes not relevant to your feature